### DISCUSSION TITLE
can the python code be separated file from the xml?

### DISCUSSION THREAD

**Discussion by xibeisiber** [JdeRobot/bt-studio#245] [Ideas]
Hi, I tried the examples and downloaded the app as ros2 package. And find the python scripts for all actions are embed in app_tree.xml.

Is it possible to generate the actions scripts as separated python files so that we can modify them more conveniently?

Thanks for your work!

**Comment by javizqh** [comment]
Right now the only way possible of modifying the raw python files is on the bt Studio editor or directly in the backend directory.

We are looking into that at the moment because of the integration with Unibotics, so this behaviour may be needed to change in the future.

Thanks for your opinion, we will take you into account in the future development.
